src/transforms/wasm.rs

```diff
@@ -0,0 +1,246 @@
+use super::Transform;
+use crate::{
+    event::Event,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+    wasm::{WasmModule, WasmModuleConfig},
+};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use vector_wasm::Role;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct WasmConfig {
+    pub module: PathBuf,
+    /// Options to be passed to the WASM module.b
+    #[serde(default)]
+    pub options: HashMap<String, serde_json::Value>,
+}
+
+impl Into<WasmModuleConfig> for WasmConfig {
+    fn into(self) -> WasmModuleConfig {
+        WasmModuleConfig::new(Role::Transform, self.module, self.options)
+    }
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<WasmConfig>("wasm")
+}
+
+#[typetag::serde(name = "wasm")]
+impl TransformConfig for WasmConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(Wasm::new(self.clone())?))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "wasm"
+    }
+}
+
+#[derive(Debug)]
+pub struct Wasm {
+    module: WasmModule,
+}
+
+impl Wasm {
+    pub fn new(config: WasmConfig) -> crate::Result<Self> {
+        let module = WasmModule::build(config)?;
+
+        Ok(Self { module })
+    }
+}
+
+impl Transform for Wasm {
+    fn transform(&mut self, event: Event) -> Option<Event> {
+        self.module
+            .process(event)
+            .map(|outputs| outputs.into_iter().next())
+            .unwrap_or(None)
```

Comment:
Let me write that down, yes, that sounds like an amazing idea. :writing_hand:

Reply:
So basically my idea with this code is that eventually I want to support `transform_stream`. This particular line 68 is basically just a safety net since we return an `LinkedList` instead of an option. This is so later we can emit any number of events per input event.